verifiers/internal/gha/provenance.go

```diff
@@ -76,100 +76,84 @@ func verifyBuilderIDLooseMatch(prov iface.Provenance, expectedBuilderID string)
 	return nil
 }
 
-func asURI(s string) string {
-	source := s
-	if !strings.HasPrefix(source, "https://") &&
-		!strings.HasPrefix(source, "git+") {
-		source = "git+https://" + source
-	}
-	if !strings.HasPrefix(source, "git+") {
-		source = "git+" + source
-	}
-
-	return source
-}
-
 // Verify source URI in provenance statement.
-func verifySourceURI(prov iface.Provenance, expectedSourceURI string, allowNoMaterialRef bool) error {
-	source := asURI(expectedSourceURI)
+func verifySourceURI(prov iface.Provenance, expectedSourceURI string) error {
```

Comment:
`--source-branch main` still passes when it should fail. Source branch (and similarly source tag) should be retrieved from the `materials` or `resolvedDependencies`. This will be empty if a sha1 was provided was provided.

Reply:
Still need support this for SLSA v0.2